packages/starknet-snap/src/rpcs/executeTxn.test.ts

```diff
@@ -0,0 +1,190 @@
+    transaction_hash: '0x123',
+    address: '0x234',
+  });
+
+  return {
+    createAccountStub,
+    executeTxnRespMock,
+    getEstimatedFeesStub,
+    getEstmatedFeesRepsMock,
+  };
+};
+
+describe('ExecuteTxn', () => {
+  let invocationExample: any;
+  const enableAuthorize = true;
+  const state = {
+    accContracts: [],
+    erc20Tokens: [],
+    networks: [STARKNET_SEPOLIA_TESTNET_NETWORK],
+    transactions: [],
+  };
+
+  const createRequestParam = (
+    chainId: constants.StarknetChainId,
+    address: string,
+    invocations: Invocations,
+    details: UniversalDetails,
+    transactionType?: TransactionType,
+  ): ExecuteTxnParams => {
+    if (transactionType) {
+      invocations.forEach((invocation) => {
+        invocation.type = transactionType;
+      });
+    }
+    const request: ExecuteTxnParams = {
+      chainId,
+      address,
+      invocations,
+      details,
+    } as ExecuteTxnParams;
+    if (enableAuthorize) {
+      request.enableAuthorize = enableAuthorize;
+    }
+    return request;
+  };
+
+  it('executes transaction correctly when account is deployed', async () => {
+    const account = await mockAccount(constants.StarknetChainId.SN_SEPOLIA);
+    invocationExample = invocationExamples.shift();
+
+    prepareMockAccount(account, state);
+    prepareConfirmDialog();
+    const { createAccountStub, executeTxnRespMock, getEstimatedFeesStub } =
+      prepareMockExecuteTxn(invocationExample.hash, true);
+
+    const request = createRequestParam(
+      state.networks[0].chainId as any,
+      account.address,
+      invocationExample.invocations,
+      invocationExample.details,
+    );
+
+    const result = await executeTxn.execute(request);
+
+    expect(result).toStrictEqual(executeTxnRespMock);
+    expect(executeTxnUtil).toHaveBeenCalledWith(
+      expect.anything(),
+      account.address,
+      account.privateKey,
+      request.invocations.map((invocation) => invocation.payload),
+      undefined,
+      expect.objectContaining({
+        maxFee: '1000000000000000',
+        nonce: undefined,
+      }),
+    );
+    expect(getEstimatedFeesStub).toHaveBeenCalled();
+    expect(createAccountStub).not.toHaveBeenCalled();
+  });
+
+  it('executes transaction correctly when account is not deployed', async () => {
+    const account = await mockAccount(constants.StarknetChainId.SN_SEPOLIA);
+    invocationExample = invocationExamples.shift();
+
+    prepareMockAccount(account, state);
+    prepareConfirmDialog();
+    const { createAccountStub, executeTxnRespMock, getEstimatedFeesStub } =
+      prepareMockExecuteTxn(invocationExample.hash, false);
+
+    const request = createRequestParam(
+      state.networks[0].chainId as any,
+      account.address,
+      invocationExample.invocations,
+      invocationExample.details,
+    );
+
+    const result = await executeTxn.execute(request);
+
+    expect(result).toStrictEqual(executeTxnRespMock);
```

Comment:
we dont need to check the result, but we have to make sure getEstimatedFees has correct param
and createAccountSpy get called with correct param